lib/puppet/type/concat_file.rb

```diff
@@ -85,40 +92,56 @@ def exists?
   end
 
   # Inherit File parameters
-  newparam(:selinux_ignore_defaults) do
-  end
+  newparam(:selinux_ignore_defaults, :boolean => true, :parent => Puppet::Parameter::Boolean)
```

Comment:
Technically no, they're not any different. It's just that setting Puppet::Property::Boolean as a parent of a property breaks the property which is why often you see the same bits copied directly into the property.